densecap/modules/BoxSampler.lua

```diff
@@ -101,8 +101,8 @@ function BoxSampler:updateOutput(input)
   -- even if it is outside the bounds or does not meet the thresholds.
   -- This is important since things will crash if we don't have at least one
   -- positive box.
-  self.pos_mask:scatter(2, target_idx, 1)
-  self.neg_mask:scatter(2, target_idx, 0)
+  self.pos_mask:typeAs(input_max_iou):scatter(2, target_idx:typeAs(input_max_iou), 1)
```

Comment:
yea. I'm going to just fix the cutorch code itself today, rather than hacking this through.